### PR TITLE
refactor(compiler): use native BigInt when calculating i18n digests

### PR DESCRIPTION
**Pull request by clydin**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

To further modernize and improve the performance of the i18n digest generation, The 64-bit aspects of the process now use the native `BigInt` instead of a custom JavaScript implementation. This removes the need for the big_integer helper code and associated tests as the code was not used anywhere else in the framework. Only the `BigInt` constructor, `BigInt.asUintN` function, and `.toString` function are currently used. `BigInt` literals can unfortunately not yet be used due to the bazel test devmode setup which compiles the TypeScript code at an EcmaScript level that does not yet support the literals.

This change plus the recent DataView refactoring results in a ~15% performance improvement for digest calculation.

Browser support information (`@angular/compiler` should only be present in an application if using JIT mode):
- BigInt constructor: https://caniuse.com/mdn-javascript_builtins_bigint_bigint
- BigInt asUintN: https://caniuse.com/mdn-javascript_builtins_bigint_asuintn
- BigInt toString: https://caniuse.com/mdn-javascript_builtins_bigint_tostring

Node.js 10.4.0 and higher also provide support (Angular officially supports `^14.20.0 || ^16.13.0 || ^18.10.0`).

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/compiler/src/i18n/digest.ts [hunk 1]
```diff
@@ -192,21 +191,19 @@ export function fingerprint(str: string): [number, number] {
...
 }
 
 export function computeMsgId(msg: string, meaning: string = ''): string {
   let msgFingerprint = fingerprint(msg);
 
   if (meaning) {
-    const meaningFingerprint = fingerprint(meaning);
-    msgFingerprint = add64(rol64(msgFingerprint, 1), meaningFingerprint);
+    msgFingerprint = BigInt.asUintN(64, msgFingerprint << BigInt(1)) |
+        ((msgFingerprint >> BigInt(63)) & BigInt(1));
```
Comment (JoostK) [inline packages/compiler/src/i18n/digest.ts:202]: The `rol64` helper function made it a bit more clear that this implements a rotate left operation; either a comment would be helpful or extract it out into a function again.

**Review by JoostK — APPROVED** [review]
Nice cleanup!

File: packages/compiler/src/i18n/digest.ts (hunk 1 above)
Comment (clydin) [reply packages/compiler/src/i18n/digest.ts:214]: Added a comment.

File: packages/compiler/src/i18n/digest.ts [hunk 2]
```diff
@@ -203,21 +202,21 @@ export function fingerprint(str: string): [number, number] {
     lo = lo ^ -0x6b5f56d8;
```
Comment (JeanMeche) [inline packages/compiler/src/i18n/digest.ts:202]: Do we know why the value here is `-0x6b5f56d8` but `0x94a0a928` in the closure code ?

Comment (clydin) [reply packages/compiler/src/i18n/digest.ts:202]: They are the same 32-bit value.
Since it is not related to the bigint changes, it was left unchanged.

**Review by AndrewKushnir — APPROVED** [review]
@clydin this change looks great 👍

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 0f10d75228c234253616fae222c53e43dd86d949.

**Pull request closed without merging**